### PR TITLE
feat: Add support for w3c options

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
This is just POC to demonstrate my vision for W3C Options support in Python client. 
If we accept it then it would be necessary to add all the supported capabilities like we did in Java, although this could be done gradually in multiple PRs.

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
cc @titusfortner @diemol

File: test/functional/android/helper/test_helper.py [hunk 1]
```diff
@@ -36,11 +32,16 @@
 class BaseTestCase:
     def setup_method(self, method) -> None:  # type: ignore
         desired_caps = desired_capabilities.get_desired_capabilities('ApiDemos-debug.apk.zip')
-        self.driver = webdriver.Remote('http://localhost:4723/wd/hub', desired_caps)
+        self.driver = webdriver.Remote(
+            'http://localhost:4723/wd/hub', options=UiAutomator2Options().load_capabilities(desired_caps)
```
Comment (KazuCocoa) [inline test/functional/android/helper/test_helper.py:36]: Seems good, overall.

One confirmation is... when a user wants to set an arbitrary automation name etc (not defined in this client one), they can use:

```
options = AppiumOptions()
options.automation_name('arbitrary name').set_capability('ownCapability', 'the value')...
...

 self.driver = webdriver.Remote('http://localhost:4723/wd/hub', options=options)
```

, correct?

Comment (mykola-mokhnach) [reply test/functional/android/helper/test_helper.py:36]: yes, that's correct

File: appium/options/common/base.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            'browserVersion',
+            PLATFORM_NAME,
+            'pageLoadStrategy',
+            'proxy',
+            'setWindowRect',
+            'timeouts',
+            'unhandledPromptBehavior',
+        ]
+    )
+    _OSS_W3C_CONVERSION = {
```
Comment (KazuCocoa) [inline appium/options/common/base.py:47]: 👍

**Review by KazuCocoa — APPROVED** [review]
Need to fix tests, otherwise lgtm 👍

File: test/functional/android/helper/test_helper.py (hunk 1 above)
Comment (mykola-mokhnach) [reply test/functional/android/helper/test_helper.py:36]: small correction to the above code:

```
options = AppiumOptions()
options.automation_name = 'arbitrary name'
options.set_capability('ownCapability', 'the value')\
    .set_capability('appium:ownCapability2', 'the value')
     ....
```

**Review by jlipps — COMMENTED** [review]
I didn't review the code but the overall  end result seems good.

File: test/functional/android/helper/test_helper.py (hunk 1 above)
Comment (KazuCocoa) [reply test/functional/android/helper/test_helper.py:36]: thank you

**Pull request merged**